Src/BootCamp.Chapter/CaesarCipher.cs

```diff
@@ -1,22 +1,243 @@
         {
-            return message;
+            // I could have done a characterOccurrences[0] and [1] but
+            // despite the fact It's more complicated I learned more about arrays.
+            var repeatedCharacters = Deconstruct2dArray(characterOccurrences, 0);
+            var numberOfOccurrences = Deconstruct2dArray(characterOccurrences, 1);
+            var highestOccurrence = ArrayOps.FindMaxValue(numberOfOccurrences);
+
+            var possibleKeys = new int[0];
+            int currentKey;
+
+            for (int i = 0; i < repeatedCharacters.Length; i++)
+            {
+                if (numberOfOccurrences[i] >= highestOccurrence - KeyAccuracy)
+                {
+                    currentKey = repeatedCharacters[i] - BaseCharacter;
+                    possibleKeys = ArrayOps.InsertLast(possibleKeys, currentKey);
+                }
+            }
+            return possibleKeys;
         }
 
-        public static string Decrypt(string message, byte shift)
+        /// <summary>
+        /// Prints decrypted messages for each possible key available in input array.
+        /// </summary>
+        /// <param name="encryptedMessage">Message to be decrypted</param>
+        public static void PrintDecryptedVariants(string encryptedMessage)
         {
-            return message;
+            bool arguementIsValid = Utils.IsStringValid(encryptedMessage);
+            if (!arguementIsValid)
+            {
+                Console.WriteLine("Argument is not valid");
+                return;
+            }
+
+            var characterOccurrences = AnalyseFrequency(encryptedMessage);
+            var possibleKeys = FindPossibleKeys(characterOccurrences);
+
+            Console.WriteLine($"{possibleKeys.Length} possible key(s) found.{Environment.NewLine}");
+            Console.WriteLine($"Adjust the bellow settings for better results!");
+            Console.WriteLine($"KeyAccuracy: {KeyAccuracy}");
+            Console.WriteLine($"Top Character: {TopCharacter}");
+            Console.WriteLine($"BaseCharacter: {BaseCharacter}");
+            Console.WriteLine();
+
+            Console.WriteLine($"Decryption results: ");
+            foreach (var key in possibleKeys)
+            {
+                var crackedMessage = Decrypt(encryptedMessage, key);
+
+                Console.WriteLine($"Key = {key} - {crackedMessage}");
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// Transfers the contents of two one dimension arrays to one bi-dimensional array.
+        /// </summary>
+        /// <param name="array1">First int one dimension input array.</param>
+        /// <param name="array2">Second int one dimension input array.</param>
+        /// <returns>A new Bi-dimensional int array created from array1 and array2.</returns>
+        private static int[][] Construct2dArray(int[] array1, int[] array2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename method 'Construct2dArray' to match pascal case naming rules, consider using 'Construct2DArray'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012201)